client/api/src/main/java/io/smallrye/graphql/client/typesafe/api/Header.java

```diff
@@ -0,0 +1,28 @@
+package io.smallrye.graphql.client.typesafe.api;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+
+/**
+ * Adds a custom header to the http requests sent out by the GraphQL Client.
+ * Annotate your method or the {@link GraphQlClientApi API} interface.
+ *
+ * The `name` is always fixed, but the value can be either a `constant`
+ * or the name of a `method` for dynamic values. It can't be both or neither.
+ * The method is always a static method, either on the interface itself
+ * or on a fully qualified class.
+ *
+ * In rare cases, you may want to pass the value of a header as a parameter;
+ * neither `constant` nor `method` will be allowed then.
+ */
+@Retention(RUNTIME)
+@Repeatable(Headers.class)
+public @interface Header {
```

Comment:
I'd suggest adding a `@Target` annotation.  IIUC, the target should include PARAMETER, METHOD, and maybe TYPE?

Reply:
done

Reply:
good point BTW!

Reply:
I've added it to the `AuthorizationHeader`, too (but not `PARAMETER`)